coldfront/core/allocation/models.py

```diff
@@ -353,6 +360,81 @@ def get_eula(self):
         else:
             return None
 
+    def activate_user(self, user, signal_sender=None):
+        """
+        Marks an `AllocationUser` as "Active". If this allocation is active, also sends `allocation_activate_user` signal.
+
+        Params:
+            user (AllocationUser): User to activate.
+            singal_sender (str): Sender for the `allocation_activate_user` signal.
+        """
+        user.status = AllocationUserStatusChoice.objects.get(name="Active")
+        user.save()
+        if self.status.name == "Active":
+            allocation_activate_user.send(sender=signal_sender, allocation_user_pk=user.pk)
+
+    def add_user(self, user, signal_sender=None):
+        """
+        Adds a user to the allocation.
+
+        If EULAs are enabled and this allocation has an associated EULA, marks the user
+            as "PendingEULA" and sends the user an email asking them to agree to the EULA.
+        Otherwise, marks the user as "Active." Also sends the `allocation_activate_user`
+            signal if the allocation status is "Active."
+
+        Params:
+            user (User): User to add.
+            singal_sender (str): Sender for the `allocation_activate_user` signal.
+        """
+        user_status = "Active"
+        is_pending_eula = ALLOCATION_EULA_ENABLE and not user.userprofile.is_pi and self.get_eula()
```

Comment:
I am a little paranoid with ambiguous order of operations:
```suggestion
        is_pending_eula = ALLOCATION_EULA_ENABLE and (not user.userprofile.is_pi) and self.get_eula()
```

Reply:
Would `ALLOCATION_EULA_ENABLE and self.get_eula() and not user.userprofile.is_pi` work for you?